types/black-engine/core/GameObject.d.ts

```diff
@@ -0,0 +1,156 @@
+export class GameObject extends MessageDispatcher {
+    static getBoundsWithPoints(points: number[], worldTransformation: Matrix, outRect?: Rectangle): Rectangle;
+    static intersects(gameObject: GameObject, point: Vector): boolean;
+    static intersectsAt(gameObject: GameObject, point: Vector, outVector?: Vector): boolean;
+    static intersectsWith(gameObject: GameObject, point: Vector): GameObject | null;
+    static findWithTag(tag: string): GameObject[] | null;
+    static findComponents(gameObject: GameObject, type: new () => Component): Component[];
+    static forEach(gameObject: GameObject, action: (arg0: GameObject) => any): void;
+    static find(name: string, node?: GameObject): GameObject;
+    static findById(id: number, node?: GameObject): GameObject;
+    constructor();
+    private mId;
+    private mName;
+    private mComponents;
+    protected mChildren: GameObject[];
+    private mX;
+    private mY;
+    private mScaleX;
+    private mScaleY;
+    protected mPivotX: number;
+    protected mPivotY: number;
+    protected mSkewX: number;
+    protected mSkewY: number;
+    protected mAnchorX: number | null;
+    protected mAnchorY: number | null;
+    protected mPivotOffsetX: number;
+    protected mPivotOffsetY: number;
+    private mRotation;
+    protected mBoundsCache: Rectangle;
+    private mLocalTransform;
+    private mWorldTransform;
+    private mWorldTransformInverted;
+    private mDirty;
+    protected mParent: GameObject;
+    private mTag;
+    private mAdded;
+    private mNumChildrenRemoved;
+    private mNumComponentsRemoved;
+    private mDirtyFrameNum;
+    private mSuspendDirty;
+    private mCollidersCache;
+    private mChildOrComponentBeenAdded;
+    private mChildrenClone;
+    private mComponentClone;
+    make(values: any): GameObject;
+    get id(): number;
+    checkStatic(includeChildren?: boolean): boolean;
+    onAdded(): void;
+    onRemoved(): void;
+    add(...gameObjectsAndOrComponents: Array<GameObject | Component>): GameObject;
+    addChild(child: GameObject): GameObject;
+    addChildAt(child: GameObject, index?: number): GameObject;
+    private __setParent;
+    setChildIndex(child: GameObject, index: number): GameObject;
+    removeFromParent(): GameObject;
+    removeChild(child: GameObject): GameObject;
+    getChildByName(name: string): GameObject | null;
+    removeChildAt(index: number): GameObject | null;
+    removeAllChildren(): GameObject;
+    getChildAt(index: number): GameObject;
+    addComponent(component: Component): Component;
+    addComponentAt(component: Component, index?: number): Component;
+    removeComponentAt(index: number): Component | null;
+    removeComponent(instance: Component): Component | null;
+    removeAllComponents(): GameObject;
+    getComponent(typeName: () => void): Component | null;
```

Comment:
This type declaration is wrong. The type of `typeName` should be typeof Component. Source is located [here](https://github.com/MassiveHeights/Black/blob/af2ae79b6134d11d473e066c552e183de04e2ed7/src/core/GameObject.js#L602).

Reply:
Fixed in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59390/commits/a9e79a681b75d32901de1381aa5127647e4f0f5e